app/services/translation_service/store/translation.rb

```diff
@@ -0,0 +1,201 @@
+module TranslationService::Store::Translation
+
+  CommunityTranslationModel = ::CommunityTranslation
+
+  Translation = EntityUtils.define_builder(
+    [:translation_key, :mandatory, :string],
+    [:locale, :mandatory, :string],
+    [:translation])
+
+
+  module_function
+
+  # Create translations
+  # Format for translation_groups:
+  # [ { translation_key: nil // optional key - if defined it will override previous translations
+  #   , translations:
+  #     [ { locale: "en-US"
+  #       , translation: "Welcome"
+  #       }
+  #     , { locale: "fi-FI"
+  #       , translation: "Tervetuloa"
+  #       }
+  #     ]
+  #   }
+  # ]
+  def create(community_id:, translation_groups: [])
+    if translation_groups.empty?
+      msg = "You must specify 'translation_groups' as an array of hash-objects containing translation_key and array of translations - like translation_groups: [ { translation_key: nil, translations: [ { locale: 'en-US' , translation: 'Hi!'}, { locale: 'fi-FI', translation: 'Moi!'}]}]"
+      raise ArgumentError.new(msg)
+    end
```

Comment:
yeah, my point was just that there is less repeated code if arguments are checked in single point instead of all the places that redirects to that one point.